native/src/modules/categories/components/CategoryListContent.js

```diff
@@ -0,0 +1,179 @@
+// @flow
+
+import React from 'react'
+import { Linking, Dimensions, View, Text } from 'react-native'
+import Html, { GestureResponderEvent, type HTMLNode, type RendererFunction } from 'react-native-render-html'
+import { _constructStyles } from 'react-native-render-html/src/HTMLStyles'
+import styled from 'styled-components/native'
+import { type StyledComponent } from 'styled-components'
+import type { NavigationStackProp } from 'react-navigation-stack'
+import type { NavigateToInternalLinkParamsType } from '../../app/createNavigateToInternalLink'
+import buildConfig from '../../app/constants/buildConfig'
+import type { ThemeType } from '../../../../build-configs/ThemeType'
+import { RTL_LANGUAGES } from '../../i18n/constants'
+
+const VerticalPadding: StyledComponent<{}, {}, *> = styled.View`
+  padding: 0 20px;
+`
+type WindowEventType = {
+  window: {width: number,
+  height: number}
+}
+
+type ContentPropsType = {|
+  content: string,
+  navigation: NavigationStackProp<*>,
+  resourceCacheUrl: string,
+  navigateToInternalLink?: NavigateToInternalLinkParamsType => void,
+  cacheDictionary: {[string]: string},
+  language: string,
+  theme: ThemeType
+|}
+
+const HIJACK = new RegExp(buildConfig().internalLinksHijackPattern)
+
+class CategoryListContent extends React.Component<ContentPropsType, {width: number}> {
+  constructor () {
+    super()
+    this.state = {
+      width: Dimensions.get('window').width
+    }
+  }
+
+  onChange = ({ window: { width } }: WindowEventType) => {
+    this.setState({ width: width })
+  };
+
+  componentDidMount () {
+    Dimensions.addEventListener('change', this.onChange)
+  }
+
+  componentWillUnmount () {
+    Dimensions.removeEventListener('change', this.onChange)
+  }
+
+  onLinkPress = (evt: GestureResponderEvent, url: string) => {
+    const { language, navigation, navigateToInternalLink } = this.props
+
+    console.log(url)
+    if (url.includes('.pdf')) {
+      navigation.navigate('PDFViewModal', { url })
+    } else if (url.includes('.png') || url.includes('.jpg')) {
+      navigation.navigate('ImageViewModal', { url })
+    } else if (navigateToInternalLink && HIJACK.test(url)) {
+      navigateToInternalLink({
+        url,
+        language
+      })
+    } else {
+      Linking.openURL(url).catch(err => console.error('An error occurred', err))
+    }
```

Comment:
Is this copied from somewhere else? Maybe it makes sense to move this in a helper method.

Reply:
Sure. Sounds reasonable.